website/versioned_docs/version-0.1.0/Connection/5.mongodb_connection.md

```diff
@@ -0,0 +1,137 @@
+
+---
+id: mongodb_connection_registry
+title: MongoDB Connection Registry
+sidebar_label: MongoDB Connection Registry
+---
+
+
+`MongoDBConnection` provide the `mongo_client` on every call to the `get` method. And will `close` the connection when exiting the `"with"` statement.
+
+## Example
+##### config.yaml
+
+```yaml
+core_lib:
+  mongodb:
+    _target_: core_lib.connection.mongodb_connection_registry.MongoDBConnectionRegistry
+    config:
+      url:
+        protocol: mongodb+srv
+        host: server.example.com
+        port: 27017
+        username: test
+        password: test123
+```
+
+##### mongodb_connection.py
+
+```python
+from pymongo import MongoClient
+from core_lib.connection.connection import Connection
+
+
+class MongoDBConnection(Connection):
+    def __init__(self, mongo_client: MongoClient):
+        self.mongo_client = mongo_client
+
+    def __enter__(self) -> MongoClient:
+        return self.mongo_client
+
+    def __exit__(self, exec_type, exec_value, traceback):
+        pass
+```
+
+#### mongodb_connection_registry.py
+
+```python
+from omegaconf import DictConfig
+import pymongo

```

Comment:
use the same import statement as in other files
`from pymongo import MongoClient`